File: app/src/main/java/dev/ragnarok/fenrir/mvp/view/IMessagesLookView.java
package dev.ragnarok.fenrir.mvp.view;

import androidx.annotation.NonNull;

import java.util.ArrayList;

import dev.ragnarok.fenrir.model.LoadMoreState;
import dev.ragnarok.fenrir.model.Message;

public interface IMessagesLookView extends IBasicMessageListView, IErrorView {

    void focusTo(int index);

    void setupHeaders(@LoadMoreState int upHeaderState, @LoadMoreState int downHeaderState);

    void forwardMessages(int accountId, @NonNull ArrayList<Message> messages);

    void showDeleteForAllDialog(ArrayList<Integer> ids);
}

File: app_fenrir/src/main/kotlin/dev/ragnarok/fenrir/mvp/view/IMessagesLookView.kt
package dev.ragnarok.fenrir.mvp.view

import dev.ragnarok.fenrir.model.LoadMoreState
import dev.ragnarok.fenrir.model.Message

interface IMessagesLookView : IBasicMessageListView, IErrorView {
    fun focusTo(index: Int)
    fun setupHeaders(@LoadMoreState upHeaderState: Int, @LoadMoreState downHeaderState: Int)
    fun forwardMessages(accountId: Int, messages: ArrayList<Message>)
}